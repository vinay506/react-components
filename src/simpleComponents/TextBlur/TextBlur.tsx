import React from 'react'
import styled from 'styled-components';
function TextBlur(props: any) {
    /**
     * Text: The text which should be displayed on web page
     * color: Defaults to darkgrey, To override the existing color we can pass Color code or color name as props.
     */
    const { text, color } = props;
    const customStyles = {
        color: color
    };
    return (
        <React.Fragment>
            <Label id="textBlur" style={customStyles}>
                {text}
            </Label>
        </React.Fragment >
    )
}

const Label = styled.div`
    color: darkgrey;
    font-weight: 400;
    font-size: 12px;
    `;
export default TextBlur
