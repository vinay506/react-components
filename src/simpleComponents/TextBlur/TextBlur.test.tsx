import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import TextBlur from './TextBlur';
import 'jest-styled-components';
describe('Text Blur', () => {
    let wrapper: any;
    let propInfo: any;
    let defaultStyles: any;
    beforeEach(() => {
        propInfo = {
            text: 'Sample Text Blur',
            color: 'red'
        };
        defaultStyles = {
            color: 'darkgrey',
            fontSize: '12px',
            fontWeight: 400
        }
        wrapper = shallow(<TextBlur {...propInfo} />);
    });
    test('Blur Text Snapshot', () => {
        const instance = renderer.create(<TextBlur {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Blur Text Element check', () => {
        const elementLength = wrapper.find('#textBlur').length;
        expect(elementLength).toEqual(1);
    })
    test('Blur Text Validation', () => {
        const primaryText = wrapper.find('#textBlur').text();
        expect(primaryText).toEqual(propInfo.text);
    })
    test('Blur Text Color Validation', () => {
        const primaryTextStyles = wrapper.find('#textBlur');
        expect(primaryTextStyles.props().style.color).toEqual(propInfo.color);
    })


    test('Text Blur Color Validation', () => {
        const instance = renderer.create(<TextBlur {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.color);
    })

    test('Text Blur font-size Validation', () => {
        const instance = renderer.create(<TextBlur {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.fontSize);
    })

    test('Text Blur Default font-weight Validation', () => {
        const instance = renderer.create(<TextBlur {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.fontWeight);
    })
})

