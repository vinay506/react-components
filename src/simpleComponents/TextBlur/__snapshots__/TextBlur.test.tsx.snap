// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Blur Blur Text Snapshot 1`] = `
.c0 {
  color: darkgrey;
  font-weight: 400;
  font-size: 12px;
}

<div
  className="c0"
  id="textBlur"
  style={
    Object {
      "color": "red",
    }
  }
>
  Sample Text Blur
</div>
`;
